### PR TITLE
Enabling flexmark boolean options

### PR DESCRIPTION
**Pull request by bmlong137**
This is really difficult to design, but I needed flexmark options available here: https://github.com/vsch/flexmark-java/blob/master/flexmark/src/main/java/com/vladsch/flexmark/parser/Parser.java

These changes allow you to specify those options with a <flexmarkParserOptions> configuration element, but it assumes the type is boolean and assumes to set it to the default value as specified in that class.

I know it works with option `LISTS_ORDERED_LIST_MANUAL_START`.  That option is required to make flexmark behave more like gitlab/github for lists.

Not sure how this should best be implemented for more generic cases.  95% of the fields are boolean though.

### CODE REVIEW

**Comment by bmlong137** [comment]
any update on including this in `master`?  does it need a rewrite?

**Comment by walokra** [comment]
I try to look into this later this week.

**Comment by walokra** [comment]
Tests pass so this doesn't seem to affect the original functionality and the Flexmark options feature can be included as it now is. Tests would be nice to have and some documentation.

**Comment by mikecapito** [comment]
Is it possible to configure all the boolean options on https://github.com/vsch/flexmark-java/wiki/Extensions?  I saw you linked to this list of options on the release notes to this merge request, but I don't see how to configure them.

(Specifically I need to change then anchor tag from id= to name=, and there appears to be a configuration option to do that ... ANCHORLINKS_SET_NAME)

Thanks!

**Comment by walokra** [comment]
@mikecapito For now the support for Flexmark options is limited to Parser options and e.g. the ANCHORLINKS_SET_NAME. Pull requests are welcome if you want to add support for those.

File: src/main/java/com/ruleoftech/markdown/page/generator/plugin/MdPageGeneratorMojo.java [hunk 1]
```diff
@@ -281,6 +301,31 @@ private int getPegdownExtensions(String extensions) {
...
+    	if (options == null)
+    		return optionsAsDataKeys;
+    	
+        for (String opt : Arrays.asList(options.split("\\s*,\\s*"))) {
+            try {
+                Field f = Parser.class.getField(opt);
+                @SuppressWarnings("unchecked")
+				DataKey<T> dataKey = (DataKey<T>)f.get(null);
+                optionsAsDataKeys.set(dataKey, dataKey.getDefaultValue(null));
+                getLog().info("Flexmark option " + opt);
```
Comment (jodygarnett) [inline src/main/java/com/ruleoftech/markdown/page/generator/plugin/MdPageGeneratorMojo.java:316]: These log statement, being at info level, has made me hesitate to upgrade to the 2.3.0 release, as the build server logs start to fill up with the same status information each time I call the plugin.

**Pull request merged**